src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseIACService.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.ons.ctp.response.casesvc.service.impl;
+
+import java.sql.Timestamp;
+import java.time.Clock;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.ctp.response.casesvc.domain.model.CaseEvent;
+import uk.gov.ons.ctp.response.casesvc.domain.model.CaseIacAudit;
+import uk.gov.ons.ctp.response.casesvc.domain.repository.CaseEventRepository;
+import uk.gov.ons.ctp.response.casesvc.domain.repository.CaseIacAuditRepository;
+import uk.gov.ons.ctp.response.casesvc.representation.CategoryDTO;
+import uk.gov.ons.ctp.response.casesvc.service.InternetAccessCodeSvcClientService;
+
+@Service
+@Slf4j
+public class CaseIACService {
+
+  private final InternetAccessCodeSvcClientService iacClient;
+  private final CaseIacAuditRepository caseIacAuditRepo;
+  private final CaseEventRepository caseEventRepo;
+  private final Clock clock;
+
+  @Autowired
+  public CaseIACService(
+      final InternetAccessCodeSvcClientService iacClient,
+      final CaseIacAuditRepository caseIacAuditRepo,
+      final CaseEventRepository caseEventRepo,
+      final Clock clock) {
+    this.iacClient = iacClient;
+    this.caseIacAuditRepo = caseIacAuditRepo;
+    this.caseEventRepo = caseEventRepo;
+    this.clock = clock;
+  }
+
+  @Transactional
+  public String generateNewCaseIACCode(final Integer casePk) {
```

Comment:
is this method not doing too much ? Th iac looks as well should be part of the contract and passed from outside and not be part of the transactional level ( cosmetic )